engine/install/ubuntu.md

```diff
@@ -120,7 +120,7 @@ from the repository.
 
     ```console
     $ sudo apt-get update
-    $ sudo apt-get install docker-ce docker-ce-cli containerd.io docker-compose-plugin
+    $ sudo apt-get install containerd.io docker-ce docker-ce-cli docker-compose-plugin
```

Comment:
Hm... I don't think the order in which packages are specified makes a difference; `apt-get` will always take the list of package you specified, then resolve dependencies between them, and base the order in which they're installed based on that.

You can see that when running both commands (without actually installing); here's the "before";

```bash
sudo apt-get install docker-ce docker-ce-cli containerd.io docker-compose-plugin
...
The following NEW packages will be installed:
  containerd.io docker-ce docker-ce-cli docker-ce-rootless-extras docker-compose-plugin docker-scan-plugin libltdl7 libslirp0 pigz slirp4netns
...
After this operation, 423 MB of additional disk space will be used.
Do you want to continue? [Y/n]
```

And here's the "after";

```bash
sudo apt-get install containerd.io docker-ce docker-ce-cli docker-compose-plugin
...
The following NEW packages will be installed:
  containerd.io docker-ce docker-ce-cli docker-ce-rootless-extras docker-compose-plugin docker-scan-plugin libltdl7 libslirp0 pigz slirp4netns
...
After this operation, 423 MB of additional disk space will be used.
Do you want to continue? [Y/n]
```

In both cases, `containerd.io` is listed before `docker-ce`. Even if they would be installed in a different order, the systemd unit defines a dependency on the `containerd` service, so it would start containerd before it starts the engine.

Reply:
I just tested this assumption on Ubuntu 22.04.
First I removed my installed docker packages with `sudo apt remove docker-ce docker-ce-cli docker-ce-rootless-extras docker-compose-plugin docker-scan-plugin containerd.io`
and `sudo systemctl daemon-reload`.
Then i tried installing it again using the two commands above.
The first (original) way of doing it using  `sudo apt-get install docker-ce docker-ce-cli containerd.io docker-compose-plugin` returned an error during docker-ce setup and the docker service didn't start: ![Screenshot from 2022-09-28 12-42-02](https://user-images.githubusercontent.com/48213077/192760552-50fd2746-bac4-49a7-9c50-e5d2a29dab68.png)
Then I uninstalled docker packages again and ran the second command `sudo apt-get install containerd.io docker-ce docker-ce-cli docker-compose-plugin` and it worked fine.

But then I repeated this test a few times, installing and uninstalling docker and running the above commands in different orders. It looks like there is a timer here that stops the service from restarting if the uninstall / install happens to close to each other. Just uninstalling and waiting a few minutes makes both the above commands work.